integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/adapters.py

```diff
@@ -0,0 +1,258 @@
+import json
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Dict, List
+
+from haystack.dataclasses import ChatMessage, ChatRole, StreamingChunk
+from transformers import AutoTokenizer, PreTrainedTokenizer
+
+from haystack_integrations.components.generators.amazon_bedrock.handlers import DefaultPromptHandler
+
+logger = logging.getLogger(__name__)
+
+
+class BedrockModelChatAdapter(ABC):
+    """
+    Base class for Amazon Bedrock chat model adapters.
+    """
+
+    def __init__(self, generation_kwargs: Dict[str, Any]) -> None:
+        self.generation_kwargs = generation_kwargs
+
+    @abstractmethod
+    def prepare_body(self, messages: List[ChatMessage], **inference_kwargs) -> Dict[str, Any]:
+        """Prepares the body for the Amazon Bedrock request."""
+
+    def get_responses(self, response_body: Dict[str, Any]) -> List[ChatMessage]:
+        """Extracts the responses from the Amazon Bedrock response."""
+        return self._extract_messages_from_response(response_body)
+
+    def get_stream_responses(self, stream, stream_handler: Callable[[StreamingChunk], None]) -> List[str]:
+        tokens: List[str] = []
+        for event in stream:
+            chunk = event.get("chunk")
+            if chunk:
+                decoded_chunk = json.loads(chunk["bytes"].decode("utf-8"))
+                token = self._extract_token_from_stream(decoded_chunk)
+                # take all the rest key/value pairs from the chunk, add them to the metadata
+                stream_metadata = {k: v for (k, v) in decoded_chunk.items() if v != token}
+                stream_chunk = StreamingChunk(content=token, meta=stream_metadata)
+                # callback the stream handler with StreamingChunk
+                stream_handler(stream_chunk)
+                tokens.append(token)
+        responses = ["".join(tokens).lstrip()]
+        return responses
+
+    def _update_params(self, target_dict: Dict[str, Any], updates_dict: Dict[str, Any]) -> None:
```

Comment:
the `self` is never used here, seems like this can be a static method `@statichmethod`